packages/core/src/components/tag/tag.tsx

```diff
@@ -50,6 +50,13 @@ export class Tag extends React.Component<ITagProps, {}> {
             </span>
         );
     }
+
+    private onRemoveClick = (e: React.MouseEvent<HTMLButtonElement>) => {
+        const { onRemove } = this.props;
+        if (onRemove) {
+            onRemove(e, this.props);
```

Comment:
The benefit is more for succinctness than anything. Note this whole thing becomes one line with `safeInvoke`:
```tsx
Utils.safeInvoke(this.props.onRemove, e, this.props);
```